### PR TITLE
Add tests for `marlowe-protocols` codecs

### PR DESCRIPTION
**Pull request by paluh**
Adds a bunch of simple tests for message serialization layer for the protocols:

  - I ported those from a "handshake" branch which we decided to reject
  - I was able to detect one bug through this test suite

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] [Test report is updated](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe/test/test-report.md) (if relevant)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: marlowe-protocols/src/Network/Protocol/Query/Codec.hs [hunk 1]
```diff
@@ -40,7 +40,9 @@ codecQuery = binaryCodec putMsg getMsg
           putResult (coerceTag tag) results
           case delimiter of
             Nothing -> putWord8 0x01
-            Just d  -> putDelimiter (coerceTag tag) d
+            Just d  -> do
+              putWord8 0x02
+              putDelimiter (coerceTag tag) d
```
Comment (jhbertra) [inline marlowe-protocols/src/Network/Protocol/Query/Codec.hs:45]: :+1:

File: marlowe-protocols/marlowe-protocols.cabal [hunk 2]
```diff
@@ -73,3 +77,39 @@ library
...
+    , marlowe
+    , marlowe-protocols
+    , plutus-tx
+    , plutus-ledger-api
+    , QuickCheck
+    , some
+    , stm
+    , tasty
+    , tasty-hunit
+    , tasty-quickcheck
```
Comment (jhbertra) [inline marlowe-protocols/marlowe-protocols.cabal:109]: We don't need to depend on `tasty` if we're using `hspec`

**Review by jhbertra — APPROVED** [review]
Tests look good, thanks for adding these!

File: marlowe-protocols/marlowe-protocols.cabal (hunk 2 above)
Comment (paluh) [reply marlowe-protocols/marlowe-protocols.cabal]: I added `-Wunused-packages` and fixed deps.

**Comment by paluh** [comment]
Thanks for quick review!

**Comment by paluh** [comment]
I went through the code (but didn't dive too much into the details) and it looks good to me.

**Pull request merged**